packages/vite/src/node/watch.ts

```diff
@@ -10,15 +10,20 @@ export function resolveChokidarOptions(
   options: WatchOptions | undefined,
 ): WatchOptions {
   const { ignored = [], ...otherOptions } = options ?? {}
+  const arraifyIgnored = arraify(ignored)
+  if (config.build.outDir) {
+    arraifyIgnored.push(
+      glob.escapePath(path.resolve(config.root, config.build.outDir)) + '/**',
+    )
+  }
```

Comment:
maybe we could construct directly the `ignored` array? It is a bit confusing that we are adding the `outDir` to `arraifyIgnored` here.
Also, TIL we support `outDir: ''`.

Reply:
You mean like below?
```ts
const ignoredList: WatchOptions['ignored'] = [
  '**/.git/**',
  '**/node_modules/**',
  '**/test-results/**', // Playwright
  glob.escapePath(config.cacheDir) + '/**',
  ...arraify(ignored),
]
if (config.build.outDir) {
  ignoredList.push(
    glob.escapePath(path.resolve(config.root, config.build.outDir)) + '/**',
  )
}
```

Reply:
Yes, but call it `ignored` and then you can use the shortcut when passing it to the watcher options.
```js
  const resolvedWatchOptions: WatchOptions = {
    ignored,
    ignoreInitial: true,
    ignorePermissionErrors: true,
```